src/client/particles.cpp

```diff
@@ -213,11 +213,12 @@ void Particle::step(float dtime)
 
 	if (m_p.animation.type != TAT_NONE) {
 		m_animation_time += dtime;
-		int frame_length_i, frame_count;
+		int* frame_length_i = new int;
 		m_p.animation.determineParams(
 				m_material.getTexture(0)->getSize(),
-				&frame_count, &frame_length_i, NULL);
-		float frame_length = frame_length_i / 1000.0;
+				NULL, frame_length_i, NULL);
+		float frame_length = *frame_length_i / 1000.0;
+		delete frame_length_i;
```

Comment:
`determineParams` doesn't set `frame_length_i` in all cases. The proper fix is to initialize it, and not to move it onto the heap.
`int frame_length_i{};`

Reply:
> `determineParams` doesn't set `frame_length_i` in all cases.

But in this case it's always set, I went trough the function and as I understand it, it's set in this case 100%

Reply:
Right now, LTO is always default ON. Only exception is `Debug` build where it's default off. It can be enabled or disabled with `-DENABLE_LTO=FALSE` or `TRUE`

Reply:
> > `determineParams` doesn't set `frame_length_i` in all cases.
> 
> But in this case it's always set, I went trough the function and as I understand it, it's set in this case 100%

Well, *technically* `m_p.animation.type` can be any other `u8`. And the compiler has to consider this.

Anyways, a heap allocation is by far more expensive than just initializing the variables.